src/util/file_reader.ts

```diff
@@ -35,6 +42,45 @@ export async function collectCustomBehaviors(
   return collectedSources;
 }
 
+async function collectGitBehaviors(gitUrl: string): Promise<FileSources> {
+  const url = gitUrl.split("git+").pop() || "";
+  const params = new URL(url).searchParams;
+  const branch = params.get("branch") || "";
+  const relPath = params.get("path") || "";
+  const urlStripped = url.split("?")[0];
+
+  const timestamp = Date.now();
+  const tmpDir = `/tmp/behaviors-repo-${timestamp}`;
```

Comment:
```suggestion
  const tmpDir = `/tmp/behaviors-repo-${crypto.randomBytes(4).toString("hex")}`;
```

Maybe just generate a dir with random values, similar to other place in this file.